src/submodules/io/csv_reader.jl

```diff
@@ -1,39 +1,22 @@
 using CSV
-import Base: parse
 
 """
-    populate_from_csv!(filename, model, agent_type; kwargs...)
+    populate_from_csv!(filename, model, agent_type, col_map; kwargs...)
 
 Populates the given `model` using data read from csv file at `filename`.
 `agent_type` is either a subtype of `AbstractAgent` also used in `model` or
 a function that returns an agent that can be added to the model. All keyword arguments
-are forwarded to `CSV.Rows`. If `agent_type` is not a function, and the `types` keyword
-argument is not specified, it is assumed that the columns in the csv file correspond
-to fields of the struct.
+are forwarded to `CSV.Rows`. `col_map` is a map from keyword arguments of 
+`agent_type` to the column index populating that argument.
 """
 function populate_from_csv!(
     filename,
     model::ABM{S,A},
-    agent_type::B;
+    agent_type::B,
+    col_map::Dict{Symbol,Int};
     kwargs...
-) where {A, B<:Type{<:A},S}
-    !haskey(kwargs, :types) && (kwargs = (kwargs..., types = collect(fieldtypes(agent_type))))
+) where {A,B<:Type{<:A},S}
     for row in CSV.Rows(filename; kwargs...)
-        add_agent_pos!(agent_type(row...), model)
+        add_agent_pos!(agent_type(; (k => row[v] for (k, v) in  col_map)...), model)
```

Comment:
What do you do if the user has not created a keyword constructor for `agent_type`?

Reply:
Currently it's required. That's what I'm trying to address with the idea I mentioned

Reply:
Sorry, didn't read when I posted this! Yes on your suggestion, not sure if second idea is worth the effort.

> If col_map is a Vector{Int}, splat the columns in the order specified by the indexes in the vector. Could also support Vector{String} or Vector{Symbol}

perhaps skip it for now and only attempt to implement if requested. Time=money ;)